### ISSUE TITLE
PIP INSTALL TA-LIB FAILED

### ISSUE THREAD

**Issue by hfzrhmn** [TA-Lib/ta-lib-python#688]
Building wheels for collected packages: TA-lib
  Building wheel for TA-lib (pyproject.toml) ... error
  error: subprocess-exited-with-error

  × Building wheel for TA-lib (pyproject.toml) did not run successfully.
  │ exit code: 1
  ╰─> [683 lines of output]
      running bdist_wheel
      running build
      running build_py
      creating build\lib.win-amd64-cpython-313\talib
      copying talib\abstract.py -> build\lib.win-amd64-cpython-313\talib
      copying talib\deprecated.py -> build\lib.win-amd64-cpython-313\talib
      copying talib\stream.py -> build\lib.win-amd64-cpython-313\talib
      copying talib\__init__.py -> build\lib.win-amd64-cpython-313\talib
      running egg_info
      writing ta_lib.egg-info\PKG-INFO
      writing dependency_links to ta_lib.egg-info\dependency_links.txt
      writing requirements to ta_lib.egg-info\requires.txt
      writing top-level names to ta_lib.egg-info\top_level.txt
      reading manifest file 'ta_lib.egg-info\SOURCES.txt'
      reading manifest template 'MANIFEST.in'
      adding license file 'LICENSE'
      adding license file 'AUTHORS'
      writing manifest file 'ta_lib.egg-info\SOURCES.txt'
      copying talib\_abstract.pxi -> build\lib.win-amd64-cpython-313\talib
      copying talib\_common.pxi -> build\lib.win-amd64-cpython-313\talib
      copying talib\_func.pxi -> build\lib.win-amd64-cpython-313\talib
      copying talib\_stream.pxi -> build\lib.win-amd64-cpython-313\talib
      copying talib\_ta_lib.c -> build\lib.win-amd64-cpython-313\talib
      copying talib\_ta_lib.pyx -> build\lib.win-amd64-cpython-313\talib
      running build_ext
      building 'talib._ta_lib' extension
      creating build\temp.win-amd64-cpython-313\Release\talib
      "C:\Program Files (x86)\Microsoft Visual Studio\2022\BuildTools\VC\Tools\MSVC\14.42.34433\bin\HostX86\x64\cl.exe" /c /nologo /O2 /W3 /GL /DNDEBUG /MD -Ic:\ta-lib\c\include -IC:\Users\mohamad.hafiz\AppData\Local\Temp\pip-build-env-6gs3ehhg\overlay\Lib\site-packages\numpy\_core\include -IC:\Users\mohamad.hafiz\AppData\Local\Programs\Python\Python313\include -IC:\Users\mohamad.hafiz\AppData\Local\Programs\Python\Python313\Include "-IC:\Program Files (x86)\Microsoft Visual Studio\2022\BuildTools\VC\Tools\MSVC\14.42.34433\include" "-IC:\Program Files (x86)\Microsoft Visual Studio\2022\BuildTools\VC\Auxiliary\VS\include" "-IC:\Program Files (x86)\Windows Kits\10\include\10.0.22621.0\ucrt" "-IC:\Program Files (x86)\Windows Kits\10\\include\10.0.22621.0\\um" "-IC:\Program Files (x86)\Windows Kits\10\\include\10.0.22621.0\\shared" "-IC:\Program Files (x86)\Windows Kits\10\\include\10.0.22621.0\\winrt" "-IC:\Program Files (x86)\Windows Kits\10\\include\10.0.22621.0\\cppwinrt" "-IC:\Program Files (x86)\Windows Kits\NETFXSDK\4.8\include\um" /Tctalib/_ta_lib.c /Fobuild\temp.win-amd64-cpython-313\Release\talib/_ta_lib.obj
      _ta_lib.c
      C:\Users\mohamad.hafiz\AppData\Local\Temp\pip-build-env-6gs3ehhg\overlay\Lib\site-packages\numpy\_core\include\numpy\npy_1_7_deprecated_api.h(14) : Warning Msg: Using deprecated NumPy API, disable it with #define NPY_NO_DEPRECATED_API NPY_1_7_API_VERSION

      "C:\Program Files (x86)\Microsoft Visual Studio\2022\BuildTools\VC\Tools\MSVC\14.42.34433\bin\HostX86\x64\link.exe" /nologo /INCREMENTAL:NO /LTCG /DLL /MANIFEST:EMBED,ID=2 /MANIFESTUAC:NO /LIBPATH:c:\ta-lib\c\lib /LIBPATH:C:\Users\mohamad.hafiz\AppData\Local\Programs\Python\Python313\libs /LIBPATH:C:\Users\mohamad.hafiz\AppData\Local\Programs\Python\Python313 /LIBPATH:C:\Users\mohamad.hafiz\AppData\Local\Programs\Python\Python313\PCbuild\amd64 "/LIBPATH:C:\Program Files (x86)\Microsoft Visual Studio\2022\BuildTools\VC\Tools\MSVC\14.42.34433\lib\x64" "/LIBPATH:C:\Program Files (x86)\Windows Kits\NETFXSDK\4.8\lib\um\x64" "/LIBPATH:C:\Program Files (x86)\Windows Kits\10\lib\10.0.22621.0\ucrt\x64" "/LIBPATH:C:\Program Files (x86)\Windows Kits\10\\lib\10.0.22621.0\\um\x64" ta_libc_cdr.lib /EXPORT:PyInit__ta_lib build\temp.win-amd64-cpython-313\Release\talib/_ta_lib.obj /OUT:build\lib.win-amd64-cpython-313\talib\_ta_lib.cp313-win_amd64.pyd /IMPLIB:build\temp.win-amd64-cpython-313\Release\talib\_ta_lib.cp313-win_amd64.lib
         Creating library build\temp.win-amd64-cpython-313\Release\talib\_ta_lib.cp313-win_amd64.lib and object build\temp.win-amd64-cpython-313\Release\talib\_ta_lib.cp313-win_amd64.exp

      build\lib.win-amd64-cpython-313\talib\_ta_lib.cp313-win_amd64.pyd : fatal error LNK1120: 339 unresolved externals
      error: command 'C:\\Program Files (x86)\\Microsoft Visual Studio\\2022\\BuildTools\\VC\\Tools\\MSVC\\14.42.34433\\bin\\HostX86\\x64\\link.exe' failed with exit code 1120
      [end of output]

  note: This error originates from a subprocess, and is likely not a problem with pip.
  ERROR: Failed building wheel for TA-lib
Failed to build TA-lib
ERROR: ERROR: Failed to build installable wheels for some pyproject.toml based projects (TA-lib)

**Comment by hfzrhmn** [comment]
I follow everything that he do, but failed. https://www.youtube.com/watch?v=jnxqu9MhBIE

I use Windows 11.

**Comment by littlebirds** [comment]
I failed to install the ta-lib wrapper today due to following error:
```shell
  /usr/bin/ld: cannot find -lta_lib: No such file or directory
      collect2: error: ld returned 1 exit status
 ```    
 It seems that the library name is wrong, instead of ta_lib it shall be ta-lib, I downloaded the repo and made typo fix in setup.py and was able to install the wrapper this way.

**Comment by mario4tier** [comment]
The breakage will be only with latest 0.6.1 release. The library file was renamed to "ta-lib" to match the package naming best practice. So "-lta-lib" instead of "-lta_lib"

I need to discuss with @mrjbq7 about how to best handle this for python.

**Comment by mario4tier** [comment]
@littelbirds Thinking a good PR would be to search for both ta_lib and ta-lib, and give priority to use ta-lib. 

Since we have new modern ways to install ta-lib in different locations, it is to be expected that users may have multiple installations... so the new "ta-lib" names will help to distinguish the "new ways" going forward.

@mrjbq7 Similarly, the binaries were renamed and re-located for windows, but found today that it is installed under "Program Files (x86)" instead of "Program Files", so I am working to fix that ASAP.

**Comment by JosephNK** [comment]
@mario4tier 
When I did "brew install ta-lib" on OSX, 0.6.1 was installed, but the installation failed with the following error.

```
ld: warning: duplicate -rpath '/Users/josephnk/.pyenv/versions/3.11.10/lib' ignored
ld: warning: duplicate -rpath '/opt/homebrew/lib' ignored
ld: library 'ta_lib' not found
clang: error: linker command failed with exit code 1 (use -v to see invocation)
error: command '/usr/bin/clang' failed with exit code 1
```

**Comment by mrjbq7** [comment]
Is that the error from brew install? Or from pip install?Looks like brew updating 0.6.1 might break some python installs until we switch library names. On Dec 24, 2024, at 11:26 PM, Joseph, NamKung ***@***.***> wrote:﻿
@mario4tier When I did "brew install ta-lib" on OSX, 0.6.1 was installed, but the installation failed with the following error.
ld: warning: duplicate -rpath '/Users/josephnk/.pyenv/versions/3.11.10/lib' ignored
ld: warning: duplicate -rpath '/opt/homebrew/lib' ignored
ld: library 'ta_lib' not found
clang: error: linker command failed with exit code 1 (use -v to see invocation)
error: command '/usr/bin/clang' failed with exit code 1

**Comment by mario4tier** [comment]
>Looks like brew updating 0.6.1 might break some python installs until we switch library names.

Correct. brew now install "ta-lib" instead of "ta_lib", so the "pip install" will fail.

If you are busy (understandable, it is the holiday!), then I can give it a shot by tomorrow for a PR on setup.py (will look for a ta-lib installed first, and if not found will fallback to ta_lib)

**Comment by adrianromanko** [comment]
` brew install ta-lib` worked.

Running with explicit linker options hangs `python -m pip install --no-cache-dir ta-lib --global-option=build_ext --global-option="-L/usr/local/Cellar/ta-lib/0.6.1/lib" --global-option="-lta-lib"
`

`python -m pip install ta-lib` fails

logs:
```bash
% python -m pip install TA-Lib
Collecting TA-Lib
  Using cached TA-Lib-0.5.1.tar.gz (369 kB)
  Installing build dependencies ... done
  Getting requirements to build wheel ... done
  Preparing metadata (pyproject.toml) ... done
Requirement already satisfied: numpy in /Users/<user>/code/sample/.venv/lib/python3.11/site-packages (from TA-Lib) (1.26.4)
Building wheels for collected packages: TA-Lib
  Building wheel for TA-Lib (pyproject.toml) ... error
  error: subprocess-exited-with-error
  
  × Building wheel for TA-Lib (pyproject.toml) did not run successfully.
  │ exit code: 1
  ╰─> [65 lines of output]
      <string>:75: UserWarning: Cannot find ta-lib library, installation may fail.
      running bdist_wheel
      running build
      running build_py
      creating build/lib.macosx-14.0-x86_64-cpython-311/talib
      copying talib/abstract.py -> build/lib.macosx-14.0-x86_64-cpython-311/talib
      copying talib/deprecated.py -> build/lib.macosx-14.0-x86_64-cpython-311/talib
      copying talib/__init__.py -> build/lib.macosx-14.0-x86_64-cpython-311/talib
      copying talib/stream.py -> build/lib.macosx-14.0-x86_64-cpython-311/talib
      running egg_info
      writing ta_lib.egg-info/PKG-INFO
      writing dependency_links to ta_lib.egg-info/dependency_links.txt
      writing requirements to ta_lib.egg-info/requires.txt
      writing top-level names to ta_lib.egg-info/top_level.txt
      reading manifest file 'ta_lib.egg-info/SOURCES.txt'
      reading manifest template 'MANIFEST.in'
      adding license file 'LICENSE'
      adding license file 'AUTHORS'
      writing manifest file 'ta_lib.egg-info/SOURCES.txt'
      copying talib/_abstract.pxi -> build/lib.macosx-14.0-x86_64-cpython-311/talib
      copying talib/_common.pxi -> build/lib.macosx-14.0-x86_64-cpython-311/talib
      copying talib/_func.pxi -> build/lib.macosx-14.0-x86_64-cpython-311/talib
      copying talib/_stream.pxi -> build/lib.macosx-14.0-x86_64-cpython-311/talib
      copying talib/_ta_lib.c -> build/lib.macosx-14.0-x86_64-cpython-311/talib
      copying talib/_ta_lib.pyx -> build/lib.macosx-14.0-x86_64-cpython-311/talib
      running build_ext
      building 'talib._ta_lib' extension
      creating build/temp.macosx-14.0-x86_64-cpython-311/talib
      clang -Wsign-compare -Wunreachable-code -fno-common -dynamic -DNDEBUG -g -fwrapv -O3 -Wall -I/usr/local/Cellar/ta-lib/0.6.1/include -I/private/var/folders/km/rmhfnxdx5jvbvxq7k6mm6dnm0000gn/T/pip-build-env-jodipdvf/overlay/lib/python3.11/site-packages/numpy/_core/include -I/Users/<user>/code/sample/.venv/include -I/usr/local/opt/python@3.11/Frameworks/Python.framework/Versions/3.11/include/python3.11 -c talib/_ta_lib.c -o build/temp.macosx-14.0-x86_64-cpython-311/talib/_ta_lib.o
      In file included from talib/_ta_lib.c:1235:
      In file included from /private/var/folders/km/rmhfnxdx5jvbvxq7k6mm6dnm0000gn/T/pip-build-env-jodipdvf/overlay/lib/python3.11/site-packages/numpy/_core/include/numpy/arrayobject.h:5:
      In file included from /private/var/folders/km/rmhfnxdx5jvbvxq7k6mm6dnm0000gn/T/pip-build-env-jodipdvf/overlay/lib/python3.11/site-packages/numpy/_core/include/numpy/ndarrayobject.h:12:
      In file included from /private/var/folders/km/rmhfnxdx5jvbvxq7k6mm6dnm0000gn/T/pip-build-env-jodipdvf/overlay/lib/python3.11/site-packages/numpy/_core/include/numpy/ndarraytypes.h:1913:
      /private/var/folders/km/rmhfnxdx5jvbvxq7k6mm6dnm0000gn/T/pip-build-env-jodipdvf/overlay/lib/python3.11/site-packages/numpy/_core/include/numpy/npy_1_7_deprecated_api.h:17:2: warning: "Using deprecated NumPy API, disable it with "          "#define NPY_NO_DEPRECATED_API NPY_1_7_API_VERSION" [-W#warnings]
         17 | #warning "Using deprecated NumPy API, disable it with " \
            |  ^
      talib/_ta_lib.c:101618:21: warning: fallthrough annotation in unreachable code [-Wunreachable-code-fallthrough]
       101618 |                     CYTHON_FALLTHROUGH;
              |                     ^
      talib/_ta_lib.c:544:34: note: expanded from macro 'CYTHON_FALLTHROUGH'
        544 |       #define CYTHON_FALLTHROUGH __attribute__((fallthrough))
            |                                  ^
      talib/_ta_lib.c:101629:21: warning: fallthrough annotation in unreachable code [-Wunreachable-code-fallthrough]
       101629 |                     CYTHON_FALLTHROUGH;
              |                     ^
      talib/_ta_lib.c:544:34: note: expanded from macro 'CYTHON_FALLTHROUGH'
        544 |       #define CYTHON_FALLTHROUGH __attribute__((fallthrough))
            |                                  ^
      talib/_ta_lib.c:102027:21: warning: fallthrough annotation in unreachable code [-Wunreachable-code-fallthrough]
       102027 |                     CYTHON_FALLTHROUGH;
              |                     ^
      talib/_ta_lib.c:544:34: note: expanded from macro 'CYTHON_FALLTHROUGH'
        544 |       #define CYTHON_FALLTHROUGH __attribute__((fallthrough))
            |                                  ^
      talib/_ta_lib.c:102038:21: warning: fallthrough annotation in unreachable code [-Wunreachable-code-fallthrough]
       102038 |                     CYTHON_FALLTHROUGH;
              |                     ^
      talib/_ta_lib.c:544:34: note: expanded from macro 'CYTHON_FALLTHROUGH'
        544 |       #define CYTHON_FALLTHROUGH __attribute__((fallthrough))
            |                                  ^
      5 warnings generated.
      clang -bundle -undefined dynamic_lookup build/temp.macosx-14.0-x86_64-cpython-311/talib/_ta_lib.o -L/usr/local/Cellar/ta-lib/0.6.1/lib -Wl,-rpath,/usr/local/Cellar/ta-lib/0.6.1/lib -lta_lib -o build/lib.macosx-14.0-x86_64-cpython-311/talib/_ta_lib.cpython-311-darwin.so
      ld: library 'ta_lib' not found
      clang: error: linker command failed with exit code 1 (use -v to see invocation)
      error: command '/usr/bin/clang' failed with exit code 1
      [end of output]
  
  note: This error originates from a subprocess, and is likely not a problem with pip.
  ERROR: Failed building wheel for TA-Lib
Failed to build TA-Lib
ERROR: ERROR: Failed to build installable wheels for some pyproject.toml based projects (TA-Lib)
```

**Comment by mrjbq7** [comment]
Can you try the newly released ta-lib-python 0.5.2, it should look for both library names.

**Comment by nikosid** [comment]
@adrianromanko , try to restart your computer, it helped for me

**Comment by Shu882** [comment]
I successfully installed it in a virtual environment with brew and pip (Python 3.12), but I failed with system Python 3.11. Could anyone help?

MBP ~ % pip install ta-lib
DEPRECATION: Loading egg at /usr/local/lib/python3.11/site-packages/atreyu_backtrader_api-0.1.0-py3.11.egg is deprecated. pip 25.1 will enforce this behaviour change. A possible replacement is to use pip for package installation. Discussion can be found at https://github.com/pypa/pip/issues/12330
Collecting ta-lib
  Downloading TA-Lib-0.5.2.tar.gz (372 kB)
  Installing build dependencies ... done
  Getting requirements to build wheel ... done
  Installing backend dependencies ... done
  Preparing metadata (pyproject.toml) ... done
Requirement already satisfied: setuptools in /usr/local/lib/python3.11/site-packages (from ta-lib) (75.6.0)
Requirement already satisfied: numpy in /usr/local/lib/python3.11/site-packages (from ta-lib) (1.23.5)
Building wheels for collected packages: ta-lib
  Building wheel for ta-lib (pyproject.toml) ... error
  error: subprocess-exited-with-error
  
  × Building wheel for ta-lib (pyproject.toml) did not run successfully.
  │ exit code: 1
  ╰─> [92 lines of output]
      /usr/local/Cellar/python@3.11/3.11.11/Frameworks/Python.framework/Versions/3.11/lib/python3.11/site-packages/setuptools/config/_apply_pyprojecttoml.py:79: SetuptoolsWarning: `install_requires` overwritten in `pyproject.toml` (dependencies)
        corresp(dist, value, root_dir)
      running bdist_wheel
      running build
      running build_py
      creating build/lib.macosx-14.0-x86_64-cpython-311/talib
      copying talib/abstract.py -> build/lib.macosx-14.0-x86_64-cpython-311/talib
      copying talib/deprecated.py -> build/lib.macosx-14.0-x86_64-cpython-311/talib
      copying talib/__init__.py -> build/lib.macosx-14.0-x86_64-cpython-311/talib
      copying talib/stream.py -> build/lib.macosx-14.0-x86_64-cpython-311/talib
      running egg_info
      writing ta_lib.egg-info/PKG-INFO
      writing dependency_links to ta_lib.egg-info/dependency_links.txt
      writing requirements to ta_lib.egg-info/requires.txt
      writing top-level names to ta_lib.egg-info/top_level.txt
      reading manifest file 'ta_lib.egg-info/SOURCES.txt'
      reading manifest template 'MANIFEST.in'
      adding license file 'LICENSE'
      adding license file 'AUTHORS'
      writing manifest file 'ta_lib.egg-info/SOURCES.txt'
      copying talib/_abstract.pxi -> build/lib.macosx-14.0-x86_64-cpython-311/talib
      copying talib/_common.pxi -> build/lib.macosx-14.0-x86_64-cpython-311/talib
      copying talib/_func.pxi -> build/lib.macosx-14.0-x86_64-cpython-311/talib
      copying talib/_stream.pxi -> build/lib.macosx-14.0-x86_64-cpython-311/talib
      copying talib/_ta_lib.c -> build/lib.macosx-14.0-x86_64-cpython-311/talib
      copying talib/_ta_lib.pyi -> build/lib.macosx-14.0-x86_64-cpython-311/talib
      copying talib/_ta_lib.pyx -> build/lib.macosx-14.0-x86_64-cpython-311/talib
      copying talib/py.typed -> build/lib.macosx-14.0-x86_64-cpython-311/talib
      running build_ext
      building 'talib._ta_lib' extension
      creating build/temp.macosx-14.0-x86_64-cpython-311/talib
      clang -Wsign-compare -Wunreachable-code -fno-common -dynamic -DNDEBUG -g -fwrapv -O3 -Wall -isysroot /Library/Developer/CommandLineTools/SDKs/MacOSX14.sdk -I/opt/X11/include -I/usr/include -I/usr/local/include -I/opt/include -I/opt/local/include -I/opt/homebrew/include -I/opt/homebrew/opt/ta-lib/include -I/usr/local/Cellar/python@3.11/3.11.11/Frameworks/Python.framework/Versions/3.11/lib/python3.11/site-packages/numpy/core/include -I/usr/local/opt/python@3.11/Frameworks/Python.framework/Versions/3.11/include/python3.11 -c talib/_ta_lib.c -o build/temp.macosx-14.0-x86_64-cpython-311/talib/_ta_lib.o
      In file included from talib/_ta_lib.c:1235:
      In file included from /usr/local/Cellar/python@3.11/3.11.11/Frameworks/Python.framework/Versions/3.11/lib/python3.11/site-packages/numpy/core/include/numpy/arrayobject.h:5:
      In file included from /usr/local/Cellar/python@3.11/3.11.11/Frameworks/Python.framework/Versions/3.11/lib/python3.11/site-packages/numpy/core/include/numpy/ndarrayobject.h:12:
      In file included from /usr/local/Cellar/python@3.11/3.11.11/Frameworks/Python.framework/Versions/3.11/lib/python3.11/site-packages/numpy/core/include/numpy/ndarraytypes.h:1948:
      /usr/local/Cellar/python@3.11/3.11.11/Frameworks/Python.framework/Versions/3.11/lib/python3.11/site-packages/numpy/core/include/numpy/npy_1_7_deprecated_api.h:17:2: warning: "Using deprecated NumPy API, disable it with "          "#define NPY_NO_DEPRECATED_API NPY_1_7_API_VERSION" [-W#warnings]
      #warning "Using deprecated NumPy API, disable it with " \
       ^
      talib/_ta_lib.c:10083:13: error: call to undeclared function 'PyDataType_ELSIZE'; ISO C99 and later do not support implicit function declarations [-Wimplicit-function-declaration]
        __pyx_r = PyDataType_ELSIZE(__pyx_v_self);
                  ^
      talib/_ta_lib.c:10096:13: error: call to undeclared function 'PyDataType_ALIGNMENT'; ISO C99 and later do not support implicit function declarations [-Wimplicit-function-declaration]
        __pyx_r = PyDataType_ALIGNMENT(__pyx_v_self);
                  ^
      talib/_ta_lib.c:10113:15: error: call to undeclared function 'PyDataType_FIELDS'; ISO C99 and later do not support implicit function declarations [-Wimplicit-function-declaration]
        __pyx_t_1 = PyDataType_FIELDS(__pyx_v_self);
                    ^
      talib/_ta_lib.c:10113:13: error: incompatible integer to pointer conversion assigning to 'PyObject *' (aka 'struct _object *') from 'int' [-Wint-conversion]
        __pyx_t_1 = PyDataType_FIELDS(__pyx_v_self);
                  ^ ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
      talib/_ta_lib.c:10134:15: error: call to undeclared function 'PyDataType_NAMES'; ISO C99 and later do not support implicit function declarations [-Wimplicit-function-declaration]
        __pyx_t_1 = PyDataType_NAMES(__pyx_v_self);
                    ^
      talib/_ta_lib.c:10134:13: error: incompatible integer to pointer conversion assigning to 'PyObject *' (aka 'struct _object *') from 'int' [-Wint-conversion]
        __pyx_t_1 = PyDataType_NAMES(__pyx_v_self);
                  ^ ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
      talib/_ta_lib.c:10151:13: error: call to undeclared function 'PyDataType_SUBARRAY'; ISO C99 and later do not support implicit function declarations [-Wimplicit-function-declaration]
        __pyx_r = PyDataType_SUBARRAY(__pyx_v_self);
                  ^
      talib/_ta_lib.c:10151:11: error: incompatible integer to pointer conversion assigning to 'PyArray_ArrayDescr *' (aka 'struct _arr_descr *') from 'int' [-Wint-conversion]
        __pyx_r = PyDataType_SUBARRAY(__pyx_v_self);
                ^ ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
      talib/_ta_lib.c:10164:13: error: call to undeclared function 'PyDataType_FLAGS'; ISO C99 and later do not support implicit function declarations [-Wimplicit-function-declaration]
        __pyx_r = PyDataType_FLAGS(__pyx_v_self);
                  ^
      talib/_ta_lib.c:10177:13: error: call to undeclared function 'PyArray_MultiIter_NUMITER'; ISO C99 and later do not support implicit function declarations [-Wimplicit-function-declaration]
        __pyx_r = PyArray_MultiIter_NUMITER(__pyx_v_self);
                  ^
      talib/_ta_lib.c:10190:13: error: call to undeclared function 'PyArray_MultiIter_SIZE'; ISO C99 and later do not support implicit function declarations [-Wimplicit-function-declaration]
        __pyx_r = PyArray_MultiIter_SIZE(__pyx_v_self);
                  ^
      talib/_ta_lib.c:10203:13: error: call to undeclared function 'PyArray_MultiIter_INDEX'; ISO C99 and later do not support implicit function declarations [-Wimplicit-function-declaration]
        __pyx_r = PyArray_MultiIter_INDEX(__pyx_v_self);
                  ^
      talib/_ta_lib.c:10216:13: error: call to undeclared function 'PyArray_MultiIter_NDIM'; ISO C99 and later do not support implicit function declarations [-Wimplicit-function-declaration]
        __pyx_r = PyArray_MultiIter_NDIM(__pyx_v_self);
                  ^
      talib/_ta_lib.c:10229:13: error: call to undeclared function 'PyArray_MultiIter_DIMS'; ISO C99 and later do not support implicit function declarations [-Wimplicit-function-declaration]
        __pyx_r = PyArray_MultiIter_DIMS(__pyx_v_self);
                  ^
      talib/_ta_lib.c:10229:11: error: incompatible integer to pointer conversion assigning to 'npy_intp *' (aka 'long *') from 'int' [-Wint-conversion]
        __pyx_r = PyArray_MultiIter_DIMS(__pyx_v_self);
                ^ ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
      talib/_ta_lib.c:10242:13: error: call to undeclared function 'PyArray_MultiIter_ITERS'; ISO C99 and later do not support implicit function declarations [-Wimplicit-function-declaration]
        __pyx_r = PyArray_MultiIter_ITERS(__pyx_v_self);
                  ^
      talib/_ta_lib.c:10242:11: error: incompatible integer to pointer conversion assigning to 'void **' from 'int' [-Wint-conversion]
        __pyx_r = PyArray_MultiIter_ITERS(__pyx_v_self);
                ^ ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
      1 warning and 17 errors generated.
      error: command '/usr/bin/clang' failed with exit code 1
      [end of output]
  
  note: This error originates from a subprocess, and is likely not a problem with pip.
  ERROR: Failed building wheel for ta-lib
Failed to build ta-lib
ERROR: ERROR: Failed to build installable wheels for some pyproject.toml based projects (ta-lib)

**Comment by mrjbq7** [comment]
It looks like you’re using an old numpy < 2

If you want that you need the ta-lib < 0.5

I haven’t back ported the fix for TA-Lib 0.6.1 underlying C library
changing the library name yet.

**Comment by Shu882** [comment]
> It looks like you’re using an old numpy < 2 If you want that you need the ta-lib < 0.5 I haven’t back ported the fix for TA-Lib 0.6.1 underlying C library changing the library name yet.

Likely a Python problem. I deleted 3.11 and installed ta-lib with Python 3.12. It worked!

**Comment by mrjbq7** [comment]
I also released 0.4.35 with support for numpy < 2 and the recent TA-Lib 0.6.1 release

**Comment by adrianromanko** [comment]
Thanks @nikosid @mrjbq7 
I figured out that the root cause was my macOS SDK which caused compilation issues for C based Python libraries,  specifically those dependent on `ta-lib` native bindings.

- Installed MacOSX15.sdk
- Updated the `SDKROOT` to explicitly point to /Library/Developer/CommandLineTools/SDKs/MacOSX15.sdk.
- Set `MACOSX_DEPLOYMENT_TARGET=15.0` to align with the SDK.
- Reinstalled Python via `pyenv` to ensure compatibility with the updated environment.
- **Restart machine** =)

 combination of dependencies works well:
```toml
[tool.poetry.dependencies]
python = "^3.11"
pandas-ta = { git = "https://github.com/twopirllc/pandas-ta.git", rev = "development" }
pandas = "^2.2.3"
ta-lib = "^0.5.2"
```

```bash
clang --version

Apple clang version 16.0.0 (clang-1600.0.26.4)
Target: x86_64-apple-darwin24.2.0
Thread model: posix
InstalledDir: /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin
```

```bash
poetry show numpy
 name         : numpy                                             
 version      : 1.26.4                                            
 description  : Fundamental package for array computing in Python 

required by
 - numba >=1.22,<2.1
 - pandas >=1.26.0
 - pandas_ta 1.26.4
 - scipy >=1.23.5,<2.3
 - ta-lib *
 - ```
